arcgis-ios-sdk-samples/Edit data/Edit feature attachments/AttachmentsTableViewController.swift

```diff
@@ -0,0 +1,174 @@
+// Copyright 2016 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class AttachmentsTableViewController: UITableViewController {
+    
+    weak var feature: AGSArcGISFeature?
+    private var attachments: [AGSAttachment] = []
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        loadAttachments()
+    }
+    
+    private func loadAttachments() {
+        
+        //show progress hud
+        SVProgressHUD.show(withStatus: "Loading attachments")
+        
+        feature?.fetchAttachments { [weak self] (attachments: [AGSAttachment]?, error: Error?) in
+            
+            //dismiss progress hud
+            SVProgressHUD.dismiss()
+            
+            guard let self = self else {
+                return
+            }
+            
+            if let error = error {
+                self.presentAlert(error: error)
+            }
+            
+            self.attachments = attachments ?? []
+            self.tableView.reloadData()
+        }
+    }
+    
+    private func deleteAttachment(_ attachment: AGSAttachment) {
+        feature?.delete(attachment) { [weak self] (error: Error?) in
+            if let error = error {
+                print(error)
+            } else {
+                self?.loadAttachments()
```

Comment:
Do you know why `loadAttachments()` is called here instead of removing the attachment from `attachments` and removing the appropriate row from the table? It seems like that would be much more efficient. Same question in `addAction()`. It is rarely a good idea to simply call `reloadData()` to add or remove a single row.